scopesim/utils.py

```diff
@@ -479,7 +479,9 @@ def find_file(filename, path=None, silent=False):
         return None
 
     if filename.startswith("!"):
-        filename = from_currsys(filename)
+        raise ValueError(f"!-string filename should be resolved upstream: "
+                         f"{filename}")
+        # filename = from_currsys(filename)
```

Comment:
The call to `from_currsys` has been removed from `find_file()`, so every caller of `find_file()` should now do the resolving of !-values themselves. But this has not been done, `SpectralSurface.__init__()` in particular does not resolve !-values.

Reply:
But it seems quite some work to properly propagate `cmd`, e.g. through this traceback:
```
../ScopeSim/scopesim/effects/surface_list.py:26: in __init__
    self.surfaces = rad_utils.make_surface_dict_from_table(tbl)
../ScopeSim/scopesim/optics/radiometry_utils.py:161: in make_surface_dict_from_table
    surf_dict[names[ii]] = make_surface_from_row(row, **tbl.meta)
../ScopeSim/scopesim/optics/radiometry_utils.py:169: in make_surface_from_row
    surface = SpectralSurface(**kwargs)
../ScopeSim/scopesim/optics/surface.py:50: in __init__
    filename = find_file(filename)
```

OTOH, at first glance it seems that `make_surface_dict_from_table()` and `make_surface_from_row()` don't do anything surface-specific. Similar functionality should exist for the `DetectorArray` for example, so if we generalize those functions, then things might become easier.